File: sort/3impbubble.cpp
#include <iostream>
using namespace std;
int main(){
     int arr[5]={7,3,9,12,11};
     int len=sizeof(arr)/sizeof(arr[0]);
     int count=0;
     for (int i=0;i<len-1;i++){
          bool swapped=false;

          for (int j=0;j<len-1-i;i++){
               if (arr[j]>arr[j+1]){
                    int temp=arr[j];
                    arr[j]=arr[j+1];
                    arr[j+1]=temp;
                    swapped=true;
               }
          }
          if (not swapped){
               break;

          }
          else{
               count+=1;
          }
     }

     for (int i=0;i<len;i++){
          cout<<arr[i]<<endl;
     }
     cout<<"no of times the outer loop runned :"<<count<<endl;
}

File: sort/3impbubble.py
import numpy as np
arr=np.array([7,3,9,12,11])

n=len(arr)
count=0
for i in range(n-1):
     swapped=False
     
     for j in range(n-i-1):
          if arr[j]>arr[j+1]:
               arr[j],arr[j+1]=arr[j+1],arr[j]
               swapped=True
               print(arr)
     if not swapped:
          break
     else:
          count+=1


print(f"the sorted array is {arr}")
print(f"number of times outer loop runed:{count}")